File: MAP(Advanced Programming Methods)/LabAssig7/LabAssig7/Repository/PrgRepository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using LabAssig7.Model.ADT;
using LabAssig7.Model.Expressions;
using LabAssig7.Model.Statements;
using LabAssig7.Model;
using System.IO;
namespace LabAssig7.Repository
{
    public class PrgRepository:IPrgRepository
    {
        private PrgState p;
        public PrgRepository(PrgState p)
        {
            this.p = p;
        }

        public PrgState getCurrentPrgState()
        {
            return this.p;
        }
        public void logPrgState()
        {
            PrgState p = this.getCurrentPrgState();

            using (StreamWriter log = File.AppendText("D:\\Visual Studio\\C#projects\\LabAssig7\\LabAssig7\\log.txt"))
            {
                
                log.WriteLine("ExeStack:\n");
                foreach (Statement st in p.Stack)
                {
                    log.WriteLine(st);
                }
                log.WriteLine("SymbolTable:\n");
                foreach (KeyValuePair<string,int> v in p.Dict )
                {
                    log.WriteLine(v.Key+"->"+v.Value);
                }
               
                log.WriteLine("Output list:");
                foreach (int v in p.Messages)
                {
                    log.WriteLine(v);
                }

                
            }
        
        }
    }
}

File: MAP(Advanced Programming Methods)/lab2v2/src/Repository/PrgRepository.java
package Repository;

import Model.PrgState;

import java.util.ArrayList;
import java.util.List;

public class PrgRepository implements IRepository {

    private List<PrgState> myList;

    public PrgRepository()
    {
        this.myList=new ArrayList<>();
    }

    public void addPrg(PrgState el)
    {
        this.myList.add(el);
    }

    @Override
    public PrgState getCurrentPrgState() {
        return myList.get(0);
    }
}
